File: 0209-minimum-size-subarray-sum/solution.c
int minSubArrayLen(int target, int* nums, int numsSize) {
    int result=INT_MAX;
    int left=0;
    int sum=0;
    for(int i=0;i<numsSize;i++)
    {
        sum+=nums[i];
        while(sum>=target)
        {
            result=(result<i+1-left)? result:i+1-left;
            sum-=nums[left];
            left++; 
        }

    }
    return result<INT_MAX ? result:0;
    
}

File: 0209-minimum-size-subarray-sum/solution.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n = nums.length;
        int r=0,l=0;
        int ans =Integer.MAX_VALUE;
        int sum=0;
        while(r<n){
            sum+=nums[r];
            while(sum>=target){
                ans=Math.min(ans,r-l+1);
                sum-=nums[l];
                l++;
            }
            r++;
        }
        return ans!=Integer.MAX_VALUE?ans:0;


    }
}
